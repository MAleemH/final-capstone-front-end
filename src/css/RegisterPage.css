* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.list_style_none {
  list-style: none;
}

.td_none {
  text-decoration: none;
}

.fieldset_border_none {
  border: none;
}

.register_body {
  background-color: #fffcf6;
}

.register_header {
  width: 90%;
  margin: auto;
  padding: 10px 0;
  height: 40px;
  position: sticky;
  top: 0;
  left: 0;
  background: transparent;
}

.register_header img {
  height: 30px;
}

.back_none {
  border: none;
}

.register_main {
  width: 40%;
  margin: auto;
}

.logo_fig {
  display: flex;
  justify-content: center;
  padding: 65px 0;
}

.eyebox img {
  width: 30px;
}

.register_main .logo_fig img {
  height: 100px;
  margin: auto;
}

.fadebreak_box {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.register_form {
  padding: 20px 0;
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 15px;
}

.register_form fieldset {
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 100%;
}

.register_form .input_name {
  width: 100%;
  color: #172b4d;
  font-size: 15px;
  padding: 15px;
  outline: none;
  border: 1px solid #d9e1f6;
  border-radius: 10px;
}

.password_box {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #d9e1f6;
  border-radius: 10px;
  padding-right: 5px;
}

.register_form .input_password {
  width: 100%;
  color: #172b4d;
  font-size: 15px;
  padding: 15px;
  outline: none;
  border: none;
  border-radius: 10px;
}

.register_form input::placeholder {
  color: #172b4d;
  font-size: 17px;
  font-weight: 400;
  line-height: 28px;
  transition: 3s;
}

.register_form input:hover::placeholder {
  transform: translate(10px, 0);
  color: #172b4d50;
}

.register_action {
  width: 20%;
  margin: auto;
}

.register_action button {
  padding: 10px 8px;
  border-radius: 10px;
  background-color: #ff9186;
  border: 1px solid #d9e1f6;
  color: #172b4d;
}

.register_main .register_main_div {
  display: flex;
  justify-content: end;
  gap: 20px;
  align-items: center;
}

.register_footer ul a {
  color: #172b4d;
}

.register_main .register_main_div a {
  color: #172b4d;
}

.register_main .register_main_div button {
  color: #172b4d;
  padding: 10px 8px;
  border-radius: 10px;
  background-color: #ff9186;
  border: 1px solid #d9e1f6;
}

.register_footer {
  width: 90%;
  margin: auto;
  position: sticky;
  bottom: 0;
  left: 0;
  background: transparent;
  padding: 10px 0;
}

.register_footer ul {
  display: flex;
  gap: 15px;
}

.password_display {
  border: none;
  background-color: transparent;
}

.rotate_breakline {
  transform: rotate(180deg);
}

@media only screen and (max-width: 1024px) {
  .register_main {
    width: 60%;
    margin: auto;
  }
}

@media only screen and (max-width: 425px) {
  .fadebreak_box {
    display: flex;
    justify-content: center;
  }

  .register_main {
    width: 80%;
    margin: auto;
  }

  .fadebreak_box img {
    width: 75px;
    margin: auto;
  }

  .fadebreak_box p {
    font-size: 14px;
  }

  .register_form .input_name {
    width: 100%;
    color: #172b4d;
    font-size: 14px;
    padding: 15px;
    outline: none;
    border: 1px solid #d9e1f6;
    border-radius: 10px;
  }

  .register_form input::placeholder {
    color: #172b4d;
    font-size: 14px;
    font-weight: 400;
    line-height: 28px;
    transition: 3s;
  }

  .register_action {
    width: 20%;
    margin: auto;
  }

  .register_main .register_main_div {
    display: flex;
    justify-content: end;
    gap: 15px;
    align-items: center;
  }

  .register_footer ul a {
    color: #172b4d;
    font-size: 14px;
  }

  .register_main .register_main_div a {
    color: #172b4d;
    font-size: 14px;
  }
}
