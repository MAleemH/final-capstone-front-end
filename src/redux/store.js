import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
