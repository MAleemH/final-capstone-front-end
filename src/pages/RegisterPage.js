import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import '../css/RegisterPage.css';
import backImg from '../img/back.png';
import eyeImg from '../img/eye.png';
import fadingBreak from '../img/fading_break.png';
import logoImg from '../img/logo.png';

function RegisterPage() {
  const [passwordType1, setPasswordType1] = useState('password');
  const [passwordType2, setPasswordType2] = useState('password');

  return (
    <div className="register_body">

      <header className="register_header">
        <nav>
          <Link to="/">
            <img src={backImg} alt="" />
          </Link>
        </nav>
      </header>

      <main className="register_main">
        <figure className="logo_fig">
          <Link to="/">
            <img src={logoImg} alt="" />
          </Link>
        </figure>

        <div className="fadebreak_box">
          <figure>
            {' '}
            <img src={fadingBreak} alt="" />
            {' '}
          </figure>
          <p> &nbsp; Register with &nbsp; </p>
          <figure>
            {' '}
            <img className="rotate_breakline" src={fadingBreak} alt="" />
            {' '}
          </figure>
        </div>

        <form action="" className="register_form">
          <fieldset className="fieldset_border_none">
            <input className="input_name" type="text" placeholder="Username" required />

            <input className="input_name" type="email" placeholder="Email" required />

            <div className="password_box">
              <input className="input_password" type={passwordType1} placeholder="Password" required />
              {passwordType1 === 'password'
                ? (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType1('text')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )
                : (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType1('password')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )}
            </div>

            <div className="password_box">
              <input className="input_password" type={passwordType2} placeholder="Re-enter Password" required />
              {passwordType2 === 'password'
                ? (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType2('text')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )
                : (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType2('password')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )}
            </div>

          </fieldset>

          <fieldset className="fieldset_border_none register_action">
            <button type="button">Submit</button>
          </fieldset>
        </form>

        <div className="register_main_div">
          <p>
            {' '}
            <Link to="/login" className="td_none">
              already registered?
            </Link>
            {' '}
          </p>
        </div>

      </main>

      <footer className="register_footer">
        <ul className="list_style_none">
          <li><Link to="/register" className="td_none"> terms </Link></li>
          <li><Link to="/register" className="td_none">  privacy </Link></li>
        </ul>
      </footer>

    </div>
  );
}

export default RegisterPage;
