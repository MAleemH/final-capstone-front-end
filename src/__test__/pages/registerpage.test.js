import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import RegisterPage from '../../pages/RegisterPage';

it('RegisterPage component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
