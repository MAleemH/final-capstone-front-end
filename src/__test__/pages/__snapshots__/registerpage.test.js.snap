// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage component renders correctly 1`] = `
<div
  className="register_body"
>
  <header
    className="register_header"
  >
    <nav>
      <button
        className="back_none"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          src="back.png"
        />
      </button>
    </nav>
  </header>
  <main
    className="register_main"
  >
    <figure
      className="logo_fig"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
    </figure>
    <div
      className="fadebreak_box"
    >
      <figure>
         
        <img
          alt=""
          src="fading_break.png"
        />
         
      </figure>
      <p>
         
         
        Register with
         
         
         
      </p>
      <figure>
         
        <img
          alt=""
          className="rotate_breakline"
          src="fading_break.png"
        />
         
      </figure>
    </div>
    <form
      action=""
      className="register_form"
    >
      <fieldset
        className="fieldset_border_none"
      >
        <input
          aria-label="Input Username"
          className="input_name"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <input
          aria-label="Input Email"
          className="input_name"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
        <div
          className="password_box"
        >
          <input
            aria-label="Input Password"
            className="input_password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <figure
            className="eyebox"
          >
            <button
              className="password_display"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                src="eye.png"
              />
            </button>
          </figure>
        </div>
        <div
          className="password_box"
        >
          <input
            aria-label="Input Password"
            className="input_password"
            onChange={[Function]}
            placeholder="Re-enter Password"
            required={true}
            type="password"
            value=""
          />
          <figure
            className="eyebox"
          >
            <button
              className="password_display"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                src="eye.png"
              />
            </button>
          </figure>
        </div>
        <select
          aria-label="Input Label"
          className="input_name"
          id="specializationId"
          onChange={[Function]}
        >
          <option
            aria-label="Input Specialization"
            value=""
          >
            Select Role
          </option>
          <option
            aria-label="Input Specialization"
            value="client"
          >
            Client
          </option>
          <option
            aria-label="Input Specialization"
            value="admin"
          >
            Admin
          </option>
        </select>
      </fieldset>
      <fieldset
        className="fieldset_border_none register_action"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </fieldset>
    </form>
    <div
      className="register_main_div"
    >
      <p>
         
        <a
          className="td_none"
          href="/login"
          onClick={[Function]}
        >
          already registered?
        </a>
         
      </p>
    </div>
  </main>
  <footer
    className="register_footer"
  >
    <ul
      className="list_style_none"
    >
      <li>
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           terms 
        </a>
      </li>
      <li>
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
            privacy 
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
